### ISSUE TITLE
bug: snap top object on guidelines

### ISSUE THREAD

**Issue by soutiagojose** [uxbox/uxbox#3157]
### Steps To Reproduce

1. Create a board
 2. I create a base object to place the guidelines at their ends
 3. I position the guide lines under their ends

### Expected behavior

The guidelines are expected to snap to the object instead of the grid lines

### Actual behavior

Currently, when trying to select the guide lines under the ends of the object, they align with the grid lines, which are not always the closest.

### Additional context

I recently started a project and noticed that my guidelines weren't snapping to objects or vector points.  An object gets the snap to object action on another object, but the guidelines, from what I noticed, aren't having that action and go straight to the snap to grid.  I don't know if it's really a bug or if this function for guide lines never existed, but if it never existed, I hope they add it soon because it's a very important function for the alignments of a layout.

A working example is in figma that prioritizes snap to object and snap to geometry on objects and vectors and also on the guidelines of the file

**Comment by EvaMarco** [comment]
Hello  @allytiago :
This is not really a bug because it was originally intended to work this way.
Now we realize that we did not take into account the case that you comment and it really makes a lot of sense.
So I have created an enhancement with high priority to modify this operation. 
You can find it here https://tree.taiga.io/project/penpot/issue/5192
Thanks for your help.

**Comment by soutiagojose** [comment]
> Hello @allytiago : This is not really a bug because it was originally intended to work this way. Now we realize that we did not take into account the case that you comment and it really makes a lot of sense. So I have created an enhancement with high priority to modify this operation. You can find it here https://tree.taiga.io/project/penpot/issue/5192 Thanks for your help.

So, at first I even thought it was a bug but I considered it not to be.
 What causes me to miss it is that at some point, I will use an element that does not have an "even" size, which does not match the grid, for example, the safety margins for instagram posts.  The media size is 1080x1350.  Its top and bottom margins are 135px.  If I want to position two guide lines in a 135px element on a 1350px high artboard, it will give a broken pixel and the penpot will not let me place the line exactly to follow the object.

 Another point of annoyance is that it doesn't even connect to the grid line that connects to the object.  Several times I had to zoom in several times for the grid to be visible and place the guide line as close to the object as there was a grid line and the guide line did not connect to it, it connected to another one nearby because there was no snap function to object.

Today I went to create an element and I needed to create a guide line over a point of a curve and the node of another curve because I needed a very precise alignment, but the penpot didn't allow it.  I couldn't even see the x and y placements of the dots.

**Comment by mdbenito** [comment]
When I try to open the referenced taiga issue I get a 404. Is there a newer one? AFAICT there is still no "snap to object" feature, correct? @madalenapmelo-kp

**Comment by madalenapmelo-kp** [comment]
@mdbenito you're right, the referenced Taiga issue was outdated, the correct reference is: https://tree.taiga.io/project/penpot/us/4245?backlog-q=snap&no-milestone=1
